msg/service.go

```diff
@@ -87,6 +91,12 @@ func Path(s string) string {
 	for i, j := 0, len(l)-1; i < j; i, j = i+1, j-1 {
 		l[i], l[j] = l[j], l[i]
 	}
+	for i, k := range l {
```

Comment:
There is no need to loop again here, this can and should be folded into the loop above.